app/src/main/java/com/cobresun/brun/pantsorshorts/repositories/impl/SharedPrefsUserDataRepository.java

```diff
@@ -19,16 +21,43 @@ public SharedPrefsUserDataRepository(Context context) {
         this.context = context;
     }
 
+    // ***** These functions exist purely to escape a current crash ***** \\
+
+    @Override
+    public boolean hasUserUpdated() {
+        SharedPreferences settings = context.getSharedPreferences(PREFS_NAME, MODE_PRIVATE);
+        return settings.getBoolean("hasUserUpdated", true);
+    }
+
+    @Override
+    public void writeHasUserUpdated(boolean userUpdated) {
+        SharedPreferences settings = context.getSharedPreferences(PREFS_NAME, MODE_PRIVATE);
+        SharedPreferences.Editor editor = settings.edit();
+        editor.putBoolean("hasUserUpdated", userUpdated);
+        editor.apply();
+    }
+
+    @Override
+    public void clearUserThreshold() {
+        int defaultThreshold = 21;
```

Comment:
I would say to make this a global variable from somewhere so it's easier to read later but it's a hacky fix anyway